mantidproject/mantid
# Fix being unable to fit on a plot with no legend

**Description of work.**
This PR fixes an issue where attempting to fit on a plot with no legend throws an error. It also makes it so that doing a fit or plotting a guess on a plot with no legend does not create a legend.

**To test:**
1. Run the following script in Workbench:
```
from mantid.simpleapi import *

import math
import matplotlib.pyplot as plt
import numpy as np
from scipy import stats

np.random.seed(101)

mu = 0
variance = 1
sigma = math.sqrt(variance)
x = np.linspace(mu - 3*sigma, mu + 3*sigma, 100)
y = stats.norm.pdf(x, mu, sigma) + np.random.randn(100)*0.005
err = 0.01*np.ones(100)

ws = CreateWorkspace(x, y, err, OutputWorkspace='ws')
fig, ax = plt.subplots(1, 1, subplot_kw={'projection': 'mantid'})
ax.errorbar(ws, specNum=1)
plt.show()
```

Plot a guess and fit and see that no error is thrown and a legend is not created.

Fixes #26974 

No release notes needed because this is bug isn't present in a release.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
